### PR TITLE
[CssSelector] Add support for :scope

### LINKED ISSUE #48773: [CssSelector] Add support for :scope selectors
### Description

https://github.com/scrapy/cssselect (the python code that was ported to create the CssSelector component) has added support for them in their 1.1.0 release.

### Example

_No response_

### PR DESCRIPTION
**Pull request by franckranaivo**
| Q             | A
| ------------- | ---
| Branch       | 6.3
| Bug fix      | no
| New feature  | yes
| Deprecations | no
| Tickets       | Fix [#48773](https://github.com/symfony/symfony/issues/48773)
| License       | MIT
| Doc PR       | 

This PR adds support for  ':scope' selector

### CODE REVIEW

File: src/Symfony/Component/CssSelector/Parser/Parser.php [hunk 1]
```diff
@@ -192,7 +192,19 @@ private function parseSimpleSelector(TokenStream $stream, bool $insideNegation =
 
                 if (!$stream->getPeek()->isDelimiter(['('])) {
                     $result = new Node\PseudoNode($result, $identifier);
-
+                    if ('Pseudo[Element[*]:scope]' === $result->__toString()) {
+                        $used = \count($stream->getUsed());
+                        if (!((2 === $used)
+                           || ((3 === $used) && $stream->getUsed()[0]->isWhiteSpace())
+                           || (($used >= 3) && $stream->getUsed()[$used - 3]->isDelimiter([',']))
+                           || (($used >= 4)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/CssSelector/Parser/Parser.php]: brackets around comparisons should be removed

File: src/Symfony/Component/CssSelector/Parser/Parser.php [hunk 2]
```diff
@@ -192,7 +192,19 @@ private function parseSimpleSelector(TokenStream $stream, bool $insideNegation =
...
+                                && $stream->getUsed()[$used - 3]->isWhiteSpace()
+                                && $stream->getUsed()[$used - 4]->isDelimiter([',']))
+                        )
+                        ) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/CssSelector/Parser/Parser.php]: should be `))) {` (putting all closing brackets together

File: src/Symfony/Component/CssSelector/Parser/Parser.php (hunk 1 above)
Comment (franckranaivo) [reply src/Symfony/Component/CssSelector/Parser/Parser.php]: okay, brackets are now removed

File: src/Symfony/Component/CssSelector/Parser/Parser.php (hunk 2 above)
Comment (franckranaivo) [reply src/Symfony/Component/CssSelector/Parser/Parser.php]: done

**Review by stof — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @franckranaivo.

**Pull request merged**